node/src/components/linear_chain.rs

```diff
@@ -444,22 +461,32 @@ where
                     self.signature_cache.insert(*signatures.clone());
                 }
                 // Check if the validator is bonded in the era in which the block was created.
-                effect_builder
-                    .is_bonded_validator(fs.era_id, fs.public_key)
-                    .map(|is_bonded| {
-                        if is_bonded {
-                            Ok((maybe_signatures, fs, is_bonded))
-                        } else {
-                            Err((maybe_signatures, fs))
-                        }
-                    })
+                // TODO: Use protocol version that is valid for the block's height.
```

Comment:
I wonder, how are we going to do that? `FinalitySignature` has only a `block_hash` and `era_id`. Do you want to find out the protocol based on the era?

Reply:
I think we could get it from the block header, couldn't we?

Reply:
I don't see `FinalitySignature` contain a block header.

Reply:
FinalitySignature doesn't, but we can use the block hash to read the block header, can't we?

Reply:
Not if we receive a finality signature for a block we don't have yet – we still might want to cache if for later b/c we expect it to come shortly.

Reply:
But we don't need the protocol version of the block that the finality signature is for.
We need the version of a block that contains the validator set of the era of the block that the finality signature is for.

Reply:
Right, that's what I meant - we have an era ID and we need to find a protocol version that is applicable for that era.

Reply:
Isn't this a blocker?

Reply:
Well, the old code used `ProtocolVersion::new(SemVer::V1_0_0)`, so it was even worse.
Also: Do we ever have to use a protocol version different from the current one?

Reply:
The old code was months old, from before we had chainspec loader and different protocol versions.

Reply:
Yes, I'm just saying maybe it isn't a blocker for this PR, because without this PR it's even worse?